### PR TITLE
Implement new PTML tags

### PR DESCRIPTION
**Pull request by mrjackv**
Sibling PRs:
* [revng-c](https://rev.ng/gitlab/revng-private/revng-c/-/merge_requests/169)
* [revng-ui](https://rev.ng/gitlab/revng-private/revng-ui/-/merge_requests/20)

### CODE REVIEW

File: python/revng/api/manager.py [hunk 1]
```diff
@@ -303,8 +303,6 @@ def _run_analysis(
             options._string_map,
         )
 
-        if result != ffi.NULL and not _api.rp_diff_map_is_empty(result):
```
Comment (aleclearmind) [inline python/revng/api/manager.py:306]: In the commit message state why we do this and what needs to be done to re-enable this behavior.

File: lib/PTML/Doxygen.cpp [hunk 2]
```diff
@@ -461,9 +478,13 @@ gatherReturnValueComments(const model::Binary &Binary,
...
         auto &Tag = Line->emplace_back(DoxygenToken::Types::Untagged,
                                        ": " + ReturnValue.Comment());
-        Tag.ExtraAttributes.emplace_back(ptml::attributes::ModelEditPath,
-                                         model::editPath::comment(*FT,
-                                                                  ReturnValue));
+        Tag.ExtraAttributes
+          .emplace_back(ptml::attributes::ActionContextLocation,
+                        serializedLocation(ranks::ReturnRegister,
+                                           FT->key(),
+                                           ReturnValue.key()));
```
Comment (aleclearmind) [inline lib/PTML/Doxygen.cpp:485]: Can you try to make a final commit that introduces a `using namespace ptml` in this `.cpp` file and maybe also introduces a `toString` local function that aliases `serializedLocation`.

If it improves the readability of the file significantly, put it in the PR.

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: python/revng/api/manager.py (hunk 1 above)
Comment (mrjackv) [reply python/revng/api/manager.py:306]: Done

File: lib/PTML/Doxygen.cpp (hunk 2 above)
Comment (mrjackv) [reply lib/PTML/Doxygen.cpp:485]: I've done the changes you asked, they didn't help much.
I can push the commit/give you the diff if you want to take a look, but it all amounted to ~20 lines __barely__ changing.

Comment (mrjackv) [reply lib/PTML/Doxygen.cpp:485]: I've done the changes you asked, they didn't help much.
I can push the commit/give you the diff if you want to take a look, but it all amounted to ~20 lines _barely_ changing.

**Comment by aleclearmind** [comment]
Merged in feef267. Thanks.

**Pull request closed without merging**